internal/concurrency/concurrency.go

```diff
@@ -0,0 +1,144 @@
+package concurrency
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/speakeasy-api/speakeasy/internal/config"
+	"github.com/speakeasy-api/speakeasy/internal/env"
+)
+
+// ErrLockExists indicates another process already holds the lock
+var ErrLockExists = errors.New("lock already exists: another Speakeasy process is running. You can: 1) wait for it to complete, 2) wait 1 minute for the lock to expire, or 3) manually remove the .lock file in your ~/.speakeasy directory if no other Speakeasy process is running")
+
+// MaxLockAge defines how old a lock can be before it's considered stale (1 minute)
+const MaxLockAge = time.Minute
+
+// AcquireLock attempts to atomically create a lock file
+// Returns nil if successful, ErrLockExists if another process holds the lock
+func AcquireLock() error {
+	if env.IsConcurrencyLockDisabled() {
+		return nil
+	}
+
+	speakeasyHomeDir, err := config.GetSpeakeasyHomeDir()
+	if err != nil {
+		return err
+	}
+
+	lockFile := filepath.Join(speakeasyHomeDir, ".lock")
```

Comment:
Nit: two suggestions:

* Explicitly name the lock file for the CLI process, in case there are other future lock file use cases in the same directory (e.g. `.cli-lock`)
* Move the lockfile name string to a constant